### PR TITLE
Add z option to all annotations

### LINKED ISSUE #175: Feature request: Annotation z-index
I've noticed that the annotation line is somehow z-indexed on the Canvas higher than the hover tooltips for the normal Chart.js line tooltips.

### PR DESCRIPTION
**Pull request by stockiNail**
Fix #175 

This PR is adding new option to all annotations and their label subelements in order to enable the user to define which annotations should be drawn on top, with the same `drawTime`.

### CODE REVIEW

File: src/annotation.js
Comment (LeeLenaleee) [inline src/annotation.js:164]: Would rename the function to something like `getDrawableElements` since in its current implementation it does not do the actual drawing

File: docs/guide/types/box.md
Comment (LeeLenaleee) [inline docs/guide/types/box.md:96]: Don't think `cresent` is the correct wording, guess it should be `ascending`. Might be my lack of english but also when I putted it through translate it didn't give the results I would have thought.

Same comment for the other files

**Review by LeeLenaleee — CHANGES_REQUESTED** [review]
2 small things

File: src/annotation.js
Comment (stockiNail) [reply src/annotation.js:164]: yeah, make a lot of sense

File: docs/guide/types/box.md
Comment (stockiNail) [reply docs/guide/types/box.md:96]: Using Google it is translated correctly but I agree to change in `ascending`

File: src/annotation.js
Comment (stockiNail) [reply src/annotation.js]: changed in getDrawableElements

File: docs/guide/types/box.md
Comment (stockiNail) [reply docs/guide/types/box.md]: Changed using `ascending`

**Comment by stockiNail** [comment]
Changing the function name, we have re-enabled code climate issue about cognitive complexity. :worried:

File: docs/guide/types/box.md
Comment (LeeLenaleee) [reply docs/guide/types/box.md]: aah yeah seems I read over it, it was at bottom of verbs, cuz all other references google thought it could be where about the moon 😅

**Review by LeeLenaleee — DISMISSED** [review]

**Comment by stockiNail** [comment]
@LeeLenaleee apologize, I have tried to solve CC issue about complexity (I think better to have an exception as was before) and your approval is dismissed.

**Comment by LeeLenaleee** [comment]
np

**Review by LeeLenaleee — APPROVED** [review]

**Comment by LeeLenaleee** [comment]
Common MD file that gets imported would be better, would be more in line with chart.js itself and will make it more user frienndly because people will only go to the place they are using and prob won't see the common page

**Comment by stockiNail** [comment]
> Common MD file that gets imported would be better, would be more in line with chart.js itself and will make it more user frienndly because people will only go to the place they are using and prob won't see the common page

OK! The content for the next PR is already decided! ;)

**Pull request merged**
